vm/ignite/src/micro_code/join.rs

```diff
@@ -16,28 +19,81 @@ use crate::{Runtime, ThreadState, VmError};
 /// # Errors
 ///
 /// * If the thread with the given ID is not found in the thread state hashmap.
-pub fn join(rt: &mut Runtime, tid: ThreadID) -> Result<()> {
-    let current_tid = rt.current_thread.thread_id;
-    let entry = rt.thread_states.entry(current_tid);
-
-    match entry {
-        Entry::Vacant(_) => Err(VmError::ThreadNotFound(current_tid).into()),
-        Entry::Occupied(mut entry) => {
-            entry.insert(ThreadState::Joining(tid));
-            Ok(())
-        }
-    }
+/// * If the operand stack is empty.
+/// * If the value on the operand stack is not an integer.
+pub fn join(mut rt: Runtime) -> Result<Runtime> {
+    let tid: i64 = rt
+        .current_thread
+        .operand_stack
+        .last()
+        .ok_or(VmError::OperandStackUnderflow)?
+        .clone()
+        .try_into()?;
+
+    let Some(mut zombie_thread) = rt.zombie_threads.remove(&tid) else {
+        // If the thread to join is not found, we need to yield control and try again
+        rt.current_thread.pc -= 1; // Decrement the program counter to re-execute the join instruction
+        let rt = yield_(rt)?;
+        return Ok(rt);
+    };
+
+    let result = zombie_thread
+        .operand_stack
+        .pop()
+        .ok_or(VmError::OperandStackUnderflow)?;
+
+    // Deallocate the zombie thread
+    drop(zombie_thread);
+
+    rt.current_thread.operand_stack.push(result);
+    Ok(rt)
 }
```

Comment:
IMPORTANT: JOIN does NOT pop values from the operand stack, since we may need to rerun this instructions multiple times and we need tid to stay on the operand stack

Reply:
This is ok, as long as I can POP the thread id loaded as usual since there's a POP after every stmt (to prevent accumulation on operand stack)